spec/jsSpecs/ares-launch.spec.js

```diff
@@ -185,30 +221,48 @@ describe(aresCmd +' --hosted(-H)', function() {
 
     it('Generate SampleApp', function(done) {
         const generateCmd = common.makeCmd('ares-generate');
-        exec(generateCmd + ` -t ${expectedTemplate.webapp} -p "id=com.sample.app" ${sampleAppPath}`, function (error, stdout) {
+        exec(generateCmd + ` -t ${expectedTemplate.webapp} -p "id=com.sample.app" ${sampleAppPath}`, function (error, stdout, stderr) {
+            if (stderr && stderr.length > 0) {
+                common.detectNodeMessage(stderr);
+            }
             expect(stdout).toContain(`Generating ${expectedTemplate.webapp}`);
             expect(error).toBeNull();
             done();
         });
     });
 
     it('Launch -H SampleApp', function(done) {
-        let result;
-        const child = exec(cmd + ` -H ${sampleAppPath}`, function (error, stdout) {
-            expect(stdout).toContain('Ares Hosted App is now running');
-            done();
+        let result = "";
+        const child = exec(cmd + ` -H ${sampleAppPath}`);
+        
+        child.stdout.on('data', function (data) {
+            process.stdout.write(data);
+            result += data;
+        });
+
+        child.stderr.on('data', function (data) {
+            if (data && data.length > 0) {
+                common.detectNodeMessage(data);
+            }
+            result += data;
+            expect(data).toBeNull();
         });
 
         setTimeout(() => {
             child.kill();
+            expect(result).toContain('Ares Hosted App is now running');
```

Comment:
Hm. I guess that if the "stdout" print unexpected string as "data" event, it will be passed finally.
For example, stdout will be 
"Ares Hosted App is now running"
"Unexpected string" 

=> will be passed in setTimeout() function, because the result string has "Ares Hosted App is now running". 

Let's have a talk by call on Monday.

Reply:
We've discussed this.